dds/idl/marshal_generator.cpp

```diff
@@ -1993,21 +1985,35 @@ bool marshal_generator::gen_struct(AST_Structure* node,
         "  return false;\n";
     } else {
       string expr;
+      expr += "  total_size += uint32_cdr_size;\n";
       for (size_t i = 0; i < fields.size(); ++i) {
-        if (i) expr += "\n    && ";
         const string field_name = fields[i]->local_name()->get_string();
         const string cond = rtpsCustom.getConditional(field_name);
         if (!cond.empty()) {
           expr += rtpsCustom.preFieldRead(field_name);
           expr += "(!(" + cond + ") || ";
         }
+        // TODO (sonndinh): When the stream ends before some fields on the
+        // reader side get their values, are they set to their default values
+        // or is this a place where try-construct comes into play?
```

Comment:
Yes, that is definitely try construct.

Reply:
Okay. I'll need to work with Clayton on this part.